Gems/Atom/Feature/Common/Code/Source/CoreLights/ProjectedShadowmapsPass.cpp

```diff
@@ -130,9 +156,32 @@ namespace AZ
                         origin.m_originInSlice[0] + sizeInInt,
                         origin.m_originInSlice[1] + sizeInInt);
                     pass->SetViewportScissor(viewport, scissor);
+                    pass->SetClearEnabled(false);
 
-                    pass->SetClearEnabled(!sliceIsCleared[origin.m_arraySlice]);
-                    sliceIsCleared[origin.m_arraySlice] = true;
+                    SliceInfo& sliceInfoItem = sliceInfo.at(origin.m_arraySlice);
+                    sliceInfoItem.m_shadowPasses.push_back(pass);
+                    sliceInfoItem.m_hasStaticShadows = sliceInfoItem.m_hasStaticShadows || it.m_isCached;
+                }
+            }
+
+            for (auto it : sliceInfo)
```

Comment:
would you want `const auto&` here to avoid copying the vector of shadow pass pointers?

Reply:
Whew. The danger of auto. Good catch.